src/main/java/bio/terra/janitor/service/iam/ProxiedAuthenticatedUserRequestFactory.java

```diff
@@ -0,0 +1,30 @@
+package bio.terra.janitor.service.iam;
+
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+/**
+ * Implementation of {link AuthenticatedUserRequestFactory} when HTTP requests enter from Apache
+ * Proxy. In this scenario, Janitor service is deployed behind Apache Proxy, and Apache Proxy
```

Comment:
This comment feels like it's truncated.

It may be worth noting why we can trust these headers to not be abused, e.g. "Apache proxy clears out any inbound values for these headers, so they are guaranteed to contain valid auth information."

Reply:
Thanks!